src/views/ObjectDataView.vue

```diff
@@ -1,350 +1,275 @@
-  const dummyGarages = generateDummyGarages(dummyBuildings);
+function injectButtonRows(nodes: PropertyNode[]): PropertyNode[] {
+  return nodes.map((node) => {
+    const { type } = node.data;
+
+    if (
+      type !== EntityType.Apartment &&
+      type !== EntityType.Commercial &&
+      type !== EntityType.Garage &&
+      type !== EntityType.Site
+    ) {
+      const buttonRow = createButtonRow(node.key, type);
+
+      node.children = injectButtonRows(node.children);
+      node.children.push(buttonRow);
+    }
 
-  dummyBuildings.forEach((building) => {
-    building.apartments = dummyApartments.filter(
-        (apartment) => apartment.buildingId === building.id,
-    );
-    building.garages = dummyGarages.filter((garage) => garage.buildingId === building.id);
+    return node;
   });
+}
+
+async function fetchPropertyTree(projectId: string): Promise<PropertyNode[]> {
+  return projectService
+    .getPropertyTree(projectId, 10, 0)
+    .then((data) => {
+      /* Every node exept Apartment, Commercial, Garge and Site nodes get an extra entry with 
+      the id and type of the parent as key and type with the isButtonRow attribute enabled. */
+      const nodesWithButtons = [
+        ...injectButtonRows(data.nodes),
+        createButtonRow(projectId, EntityType.Project),
+      ];
 
-  objectData.value = properties.map((property) => ({
-    ...property,
-    buildings: dummyBuildings.filter((building) => building.propertyId === property.id),
-  }));
+      objectData.value = nodesWithButtons;
+
+      return nodesWithButtons;
+    })
+    .catch((err) => {
+      error.value = `Failed to fetch object data: ${err.message || 'Unknown error'}`;
+      return [];
+    });
 }
 
 onMounted(() => {
-  fetchProperties(props.projectId)
-      .then((properties) => {
-        if (properties.length > 0) {
-          generateDummyData(properties);
-        }
-      })
-      .finally(() => {
-        isLoading.value = false;
-      });
+  fetchPropertyTree(props.projectId).finally(() => {
+    isLoading.value = false;
+  });
 });
 
-const expandAll = () => {
-  expandedRows.value = objectData.value.reduce((acc: Record<string, boolean>, property) => {
-    if (property?.id) {
-      acc[property.id] = true;
+const completeEntityAction = (entity: EntityType, action: string, entityId?: string) => {
+  if (action === 'edit') {
+    router.push({
+      path: `/project/${props.projectId}/${entity}/${entityId}`,
+    });
+  }
+  if (entityId && action === 'delete') {
+    deleteObject(entity, entityId);
+  }
+  if (action === 'create') {
+    router.push({
+      path: `/project/${props.projectId}/${entity}/create`,
+      query: { parentId: entityId }, // in this case entityId is the Id of the parent
+    });
+  }
+};
 
-      property.buildings?.forEach((building) => {
-        if (building?.id) expandedSubRows.value[building.id] = true;
-      });
-    }
-    return acc;
-  }, {});
+const deleteObject = (entity: EntityType, entityId: string) => {
```

Comment:
These todo-comments need implementing but I agree that this is not part of your issue. Do you think it'd be sensible to create a new issue for this (if it is not already covered by #337).